war-wrapper/src/main/java/org/ow2/proactive/scheduler/util/WarWrapper.java

```diff
@@ -0,0 +1,217 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.util;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.KeyException;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Options;
+import org.apache.log4j.Logger;
+import org.objectweb.proactive.core.config.CentralPAPropertyRepository;
+import org.objectweb.proactive.core.util.wrapper.BooleanWrapper;
+import org.objectweb.proactive.utils.JVMPropertiesPreloader;
+import org.ow2.proactive.authentication.crypto.Credentials;
+import org.ow2.proactive.resourcemanager.authentication.RMAuthentication;
+import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
+import org.ow2.proactive.resourcemanager.exception.RMException;
+import org.ow2.proactive.resourcemanager.frontend.ResourceManager;
+import org.ow2.proactive.scheduler.SchedulerFactory;
+import org.ow2.proactive.scheduler.common.Scheduler;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+import org.ow2.proactive.wrapper.WrapperWebConfiguration;
+
+
+/**
+ * Created by root on 11/03/17.
+ */
+public class WarWrapper extends SchedulerStarter {
+
+    private static final Logger logger = Logger.getLogger(WarWrapper.class);
+
+    private static final String SCHEDULER_RM_DEFAULT_HOME = ".";
+
+    private static String schedulerUri;
+
+    private static CommandLine cmd = null;
+
+    public static void main(String[] args) {
+        try {
+
+            WarWrapper warWrapper = new WarWrapper();
+            warWrapper.launchProactiveServer();
+
+            Thread.sleep(5000);
+
+            warWrapper.shutdownProactive();
+            logger.info("Proactive stopped");
+
+        } catch (InterruptedException e) {
+            logger.error("Error while stopping Proactive " + e.getMessage());
+        }
+    }
+
+    //********************************************Launching ProActive *************************************************
+    public void launchProactiveServer() {
+
+        String[] args = { "--no-rest" };
+
+        configureSchedulerAndRMAndPAHomes();
+        configureSecurityManager();
+        configureLogging();
+        configureDerby();
+
+        args = JVMPropertiesPreloader.overrideJVMProperties(args);
+
+        Options options = getOptions();
+
+        try {
+            cmd = getCommandLine(args, options);
+
+            start(cmd);
+
+            // the variable set by Jetty Starter
+            if (null != rmURL)
```

Comment:
Curly brackets are added, refactoring is done (a method configureURLs() is added).

The properties (in the method  configureURLs()) are obtained after starting RM and the Scheduler.
They are previously configured by JettyStarter, which is executed after starting RM and the Scheduler (so they have no impcat on RM and the Scheduler). Since JettyStarter is disabled, I just configured the same properties  (as JettyStarter) to preserve the same system configuration.